### PR TITLE
Add WIC CSV Data

### PR DESCRIPTION
**Pull request by brickman1444**
Based on #26 

This change adds a script to read a PDF of WIC (https://www.fns.usda.gov/wic/wic-fact-sheet) usage data in January 2021 provided by GCFD and save it as a CSV. The PDF is parsed with pdfplumber (https://pypi.org/project/pdfplumber/). An example of the outputted CSV data is checked at tests/resources/test_wic_expected_output.txt.

This changes makes the main script reparse the pdf every time it's run. Parsing the PDF takes about 13 seconds on my machine. I tried to optimize it more, but almost all the time is spent in Page.extract_text() and I don't think we can get around using that function without risking requiring future changes if the PDF format changes in the future.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -68,13 +70,21 @@ def main(geo_ls=["zip", "county"]):
     and save jsons to file
     '''
     # from src import dict_merge
+
+    print("Reading WIC Data")
+    wic_start_time = time.time()
+    src.wic.read_wic_data()
+    print("Reading WIC Data took: {0:.2f} seconds".format(time.time() - wic_start_time))
```
Comment (brickman1444) [inline src/main.py:77]: Do we want this sort of logging to be called from main(), or should it be done within the specific script?
Do we want to add options to turn verbose logging on or off?

File: src/wic.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import re
+import pdfplumber
+import pandas as pd
+
+
+'''
+GCFD staff provided a pdf IWIC.pdf which contains WIC data for 96 out of 102 IL Counties. We do not know why some counties
```
Comment (brickman1444) [inline src/wic.py:7]:
```suggestion
GCFD staff provided a pdf illinois_wic_data_january_2021.pdf which contains WIC data for 96 out of 102 IL Counties. We do not know why some counties
```

File: src/main.py [hunk 3]
```diff
@@ -68,13 +70,21 @@ def main(geo_ls=["zip", "county"]):
     and save jsons to file
     '''
     # from src import dict_merge
+
+    print("Reading WIC Data")
+    wic_start_time = time.time()
+    src.wic.read_wic_data()
```
Comment (brickman1444) [inline src/main.py:76]: Do we want to start passing options to the main script to opt in or out of reading specific data sources?

File: src/wic.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    # LA        Total
+
+    # We want to combine WIC1 and WIC2 columns into new column called WIC which will contain values such as "Total Women"
+    # "Total Children" "Total Infants" and "LA Total"
+    data.insert(2, "WIC", (data["WIC1"] + " " + data["WIC2"]))
+
+    # delete WIC1 and WIC 2 columns
+    data.drop(['WIC1', 'WIC2'], axis=1, inplace=True)
+
+    data.to_csv(destination_csv_filepath, index=False)
```
Comment (brickman1444) [inline src/wic.py:71]: Right now we take the dataframe and just export it to CSV. How does this data need to be integrated with the Census data?

File: src/main.py (hunk 3 above)
Comment (michaelpkuhn) [reply src/main.py:76]: For this specific data source, lets check if the output file already exists and avoid running it again if it does.

File: src/main.py (hunk 1 above)
Comment (michaelpkuhn) [reply src/main.py:77]: Yeah, I don't think it should be on by default

File: src/wic.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+def read_wic_data():
+    # PDF has 97 pages. Skip page 0 because it shows Statewide totals which we don't need
+    parse_wic_pdf("data_folder/illinois_wic_data_january_2021.pdf", "final_jsons/wic.csv", 1, 96)
+
+
+def parse_wic_pdf(source_pdf_filepath: str, destination_csv_filepath: str, first_page_zero_indexed: int, last_page_zero_indexed: int):
+    # use regular expression matching to find lines that start with certain words and save as variables:
+    # find rows that start with Total (this includes Total Women, Total Infant and Total Children rows)
+    Total_re = re.compile("Total")
```
Comment (michaelpkuhn) [inline src/wic.py:20]: Variables should be lower case. We're using Flake8 as the linter.

File: src/wic.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                    # maxsplit=1 because some counties have spaces in their name, example: Jo Daviess
+                    County = (line.split(sep=" ", maxsplit=1)) 
+                elif Total_re.match(line):
+                    new_line = (line.split(sep=" ")) # This looks something like ["Total", "Women", 1, 2, 3, 4]
+                    new_line = County + new_line
+                    data = data.append(pd.Series(new_line, index=data.columns), ignore_index=True)
+                elif County_Total_re.match(line):
+                    new_line = (line.split(sep=" ")) # This looks something like ["LA", "Total", 1, 2, 3, 4]
+                    new_line = County + new_line
+                    data = data.append(pd.Series(new_line, index=data.columns), ignore_index=True)
```
Comment (michaelpkuhn) [inline src/wic.py:56]: I guess it works, but I wouldn't overwrite the variable here like this. For a list, append returns None, so it is a bit confusing what happens here.

Looks like pd.concat might be a better function for this.

File: src/wic.py (hunk 4 above)
Comment (michaelpkuhn) [reply src/wic.py:71]: The index needs to be the County FIPs to merge in

File: requirements.txt [hunk 7]
```diff
@@ -16,3 +16,5 @@ geojson==2.5.0
 python-dotenv==0.15.0
 # memory profiling
 guppy3==3.1.0
+# pdf parsing
+pdfplumber==0.5.27
```
Comment (michaelpkuhn) [inline requirements.txt:20]: Would take some refactoring but maybe there's a faster pdf extractor?

**Review by michaelpkuhn — CHANGES_REQUESTED** [review]

**Review by michaelpkuhn — APPROVED** [review]

**Pull request merged**